web/scripts/dev/vite.ts

```diff
@@ -0,0 +1,41 @@
+import { createServer, type ViteDevServer } from "vite"
+import { error } from "../pretty-logs"
+import { isBuild, isClean, isServe } from "./util"
+
+export class Vite {
+  declare instance: ViteDevServer
+
+  stopped = false
+
+  static async clean() {
+    // TODO: somehow check if Vite is running?
```

Comment:
Recording pid on vite start? Checking the `3000` port?

Reply:
Vite isn't running another process. It's being ran directly. If Vite is still running, that's because there is _another_ `pnpm dev` running. I could check the port but I wouldn't be able to get the PID, I don't think.